algorithms/sorting/merge_sort.py

```diff
@@ -1,15 +1,20 @@
-from typing import Generic, TypeVar, Callable
-from operator import lt
+from algorithms.sorting.sort import Sort, T
 
-T = TypeVar('T')
 
-class MergeSort(Generic[T]):
-    def __init__(self, lst: list[T], comp_func: Callable[[T, T], bool] = lt) -> None:
-        self.comp_func = comp_func
-        self.lst = lst
+class MergeSort(Sort):
+    def _merge(self, left: list[T], right: list[T]) -> list[T]:
+        sorted_lst = []
+        while left and right:
+            next_item = left.pop(0) if self.comp_func(left[0], right[0]) else right.pop(0)
+            sorted_lst.append(next_item)
+        sorted_lst.extend(left)
+        sorted_lst.extend(right)
+        return sorted_lst
 
 
     def sort(self, lst: list[T] = None) -> list[T]:
+        super().sort()
```

Comment:
why? If removing this is not possible because of `abc` just remove `sort` in baseclass

Reply:
same for the other algorithms ofc